modules/mono/glue/GodotSharp/GodotSharp/Core/Bridge/ScriptManagerBridge.cs

```diff
@@ -427,10 +427,18 @@ static void LookupScriptForClass(Type type)
             // This method may be called before initialization.
             if (NativeFuncs.godotsharp_dotnet_module_is_initialized().ToBool() && Engine.IsEditorHint())
             {
-                foreach (var scriptPath in _pathTypeBiMap.Paths)
+                string[] script_paths = _pathTypeBiMap.Paths.ToArray();
+                if (script_paths.Length > 0)
                 {
-                    using godot_string nativeScriptPath = Marshaling.ConvertStringToNative(scriptPath);
-                    NativeFuncs.godotsharp_internal_editor_file_system_update_file(nativeScriptPath);
+                    godot_packed_string_array script_paths_packed_array = Marshaling.ConvertSystemArrayToNativePackedStringArray(script_paths);
+                    try
+                    {
+                        NativeFuncs.godotsharp_internal_editor_file_system_update_files(ref script_paths_packed_array);
+                    }
+                    finally
+                    {
+                        script_paths_packed_array.Dispose();
+                    }
```

Comment:
Just use an `using` statement.

```suggestion
                    using godot_packed_string_array scriptPaths = Marshaling.ConvertSystemArrayToNativePackedStringArray(script_paths);
                    NativeFuncs.godotsharp_internal_editor_file_system_update_files(ref scriptPaths);
```

Reply:
I cannot because the variable is passed as ref.